assets/src/edit-story/components/panels/publish/index.js

```diff
@@ -0,0 +1,246 @@
+`;
+
+const MediaWrapper = styled.div`
+  flex-basis: 134px;
+`;
+
+const BoxedText = styled.div.attrs({ role: 'button', tabIndex: '0' })`
+  color: ${({ theme }) => theme.colors.fg.v1};
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
+  display: flex;
+  flex-direction: row;
+  background-color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.1)};
+  flex: 1;
+  padding: 2px;
+  border-radius: 4px;
+  :focus {
+    outline: -webkit-focus-ring-color auto 5px;
+  }
+`;
+
+const DateWrapper = styled.div`
+  padding: 5px 0 5px 2px;
+`;
+
+const Date = styled.span`
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.86)};
+`;
+
+const Time = styled.span`
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.4)};
+`;
+
+const DateTimeWrapper = styled.div`
+  position: relative;
+`;
+
+const StyledToggleIcon = styled(ToggleIcon)`
+  height: 26px;
+  flex: 1;
+`;
+
+function PublishPanel() {
+  const {
+    state: { users },
+  } = useInspector();
+
+  const {
+    state: {
+      meta: { isSaving },
+      story: { author, date, featuredMediaUrl, publisherLogoUrl },
+      capabilities,
+    },
+    actions: { updateStory },
+  } = useStory();
+
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const dateTimePickerNode = useRef();
+  const dateFieldNode = useRef();
+
+  useKeyDownEffect(
+    dateFieldNode,
+    { key: ['space', 'enter'] },
+    () => {
+      setShowDatePicker(!showDatePicker);
+    },
+    [showDatePicker]
+  );
+
+  useFocusOut(dateTimePickerNode, () => setShowDatePicker(false), [
+    showDatePicker,
+  ]);
+
+  const handleDateChange = useCallback(
+    (value, close = false) => {
+      if (close && showDatePicker) {
+        setShowDatePicker(false);
+      }
+      updateStory({ properties: { date: value } });
+    },
+    [showDatePicker, updateStory]
+  );
+
+  const handleChangeCover = useCallback(
+    (image) =>
+      updateStory({
+        properties: {
+          featuredMedia: image.id,
+          featuredMediaUrl: image.sizes?.medium?.url || image.url,
+        },
+      }),
+    [updateStory]
+  );
+
+  // @todo Enforce square image while selecting in Media Library.
+  const handleChangePublisherLogo = useCallback(
+    (image) => {
+      updateStory({
+        properties: {
+          publisherLogo: image.id,
+          publisherLogoUrl: image.sizes?.thumbnail?.url || image.url,
+        },
+      });
+    },
+    [updateStory]
+  );
+
+  const handleChangeValue = useCallback(
+    (prop) => (value) => updateStory({ properties: { [prop]: value } }),
+    [updateStory]
+  );
+
+  const authorLabel = __('Author', 'web-stories');
+  return (
+    <SimplePanel name="publishing" title={__('Publishing', 'web-stories')}>
+      <Row>
+        <FieldLabel>{__('Publish', 'web_stories')}</FieldLabel>
```

Comment:
This label field should have correct aria and clicking on should open the date time picker. Even if dont do this now, maybe add a todo for later.

Reply:
Will create an issue for it, this seems to be a general issue for the `Label` component.